lib/product_list/cubits/product_list_cubit.dart

```diff
@@ -28,4 +29,18 @@ class ProductListCubit extends Cubit<ProductListState> {
       ));
     }
   }
+
+  Future<void> loadProductsfromIds({required List<String> productIds}) async {
+    for (var id in productIds) {
+      GetIt.instance
+          .get<ProductApi>()
+          .fetchProduct(productId: id, fields: LIST_PARAMS)
+          .then((value) {
```

Comment:
First time I'm seeing then vs. await, is this on purpose?

Reply:
Right, so initially I went with `Future.wait()` but it waits for ALL the requests to be completed to save ALL products in state.

I didn't want one request to block the display of the whole carousel so I went with this "fire and forget" implementation.